internal/store/root/move.go

```diff
@@ -124,13 +117,21 @@ func (r *Store) move(ctx context.Context, from, to string, del bool) error {
 	return nil
 }
 
-func (r *Store) moveFromTo(ctx context.Context, subFrom *leaf.Store, from, to, fromPrefix string, srcIsDir, dstIsDir, del bool) error {
+func (r *Store) moveFromTo(ctx context.Context, subFrom *leaf.Store, from, to, fromPrefix string, del bool) error {
 	ctx = ctxutil.WithGitCommit(ctx, false)
 
+	// source is a directory and not a "shadowed" leaf
+	srcIsDir := r.IsDir(ctx, from) && !r.Exists(ctx, from)
+	dstIsDir := r.IsDir(ctx, to)
+
+	if srcIsDir && r.Exists(ctx, to) && !dstIsDir {
+		return fmt.Errorf("destination is a file")
+	}
+
 	entries := []string{from}
 	// if the source is a directory we enumerate all it's children
 	// and move them one by one.
-	if r.IsDir(ctx, from) {
+	if srcIsDir {
```

Comment:
This was the culprit, but since we didn't really use `moveFromTo` anywhere else, I figured I could as well change its signature since these two variables weren't being used above in `move`